docs/reference/data-management/migrate-index-allocation-filters.asciidoc

```diff
@@ -0,0 +1,137 @@
+To change the role of an existing node in an {ess} deployment, use the
+{cloud}/ec-api-deployment-crud.html#ec_update_a_deployment[Update deployment API].
+Replace the node's `node_type` configuration with the appropriate `node_roles`.
+For example, the following configuration adds the node to the hot and content
+tiers, and enables it to act as an ingest node, remote, and transform node.
+
+[source,yaml]
+----
+"node_roles": [
+  "data_hot",
+  "data_content",
+  "ingest",
+  "remote_cluster_client",
+  "transform"
+],
+----
+
+If you are directly managing your own cluster,
+configure the appropriate roles for each node in `elasticsearch.yml`.
+For example, the following setting configures a node to be a data-only
+node in the hot and content tiers.
+
+[source,yaml]
+----
+node.roles [ data_hot, data_content ]
+----
+
+[discrete]
+[[remove-custom-allocation-settings]]
+=== Remove custom allocation settings from your {ilm-init} policy
+
+Update the allocate action for each phase to remove the attribute-based
+allocation settings. If the allocate action does not set the number of replicas,
+remove the allocate action entirely. (An empty allocate action is invalid.)
+
+IMPORTANT: The policy must specify the corresponding phase for each data tier in
+your architecture.
+For example, if you enable the warm and cold data tiers for a deployment, your
+policy must include the hot, warm, and cold phases.
+This is necessary for {ilm-init} to automatically inject the <<ilm-migrate,
+migrate>> action to move indices through the data tiers,
+even if you don't need to perform any other actions and the phase is empty.
+
+[discrete]
+[[stop-setting-custom-hot-attribute]]
+=== Stop setting the custom hot attribute on new indices
+
+When you create a data stream, its first backing index
+is now automatically assigned to `data_hot` nodes.
+Similarly, when you directly create an index, it
+is automatically assigned to `data_content` nodes.
+
+On {ess} deployments, remove the `cloud-hot-warm-allocation-0` index template
+that set the hot shard allocation attribute on all indices.
+
+[source,console]
+----
+DELETE _template/.cloud-hot-warm-allocation-0
+----
+
+If you're using a custom index template, update it to remove the node attribute
+you used to assign new indices to the hot tier.
+
+[discrete]
+[[set-tier-preference]]
+=== Set a tier preference for existing indices.
+
+{ilm-init} automatically transitions managed indices through the available
+data tiers by automatically injecting a <<ilm-migrate,migrate action>>
+into each phase.
+
+To enable {ilm-init} to move an _existing_ managed index
+through the data tiers, update the index settings to:
+
+. Remove the custom allocation filter by setting it to `null`.
+. Set the `_tier_preference`.
+
+For example, if your old template set the `data` attribute to `hot`
+to allocate shards to the hot tier, set the `data` attribute to `null`
+and set the `_tier_preference` to `data_hot`.
```

Comment:
At the time of migration, it's possible/likely that some indices are quite far away in their ILM lifecycle (in the `warm` or `cold` phase already). These indices would have the `warm` data attribute configured by the "old style" `allocate` action from the `warm` ILM phase.

I think it'd be valuable to include an example for the `warm` data attribute in the context of indices that already transitioned to the `warm` phase in their current ILM lifecycles at the time of migration. This example would also illustrate the corresponding `_tier_preference` configuration `"data_warm,data_hot"`